app/src/main/java/com/babylon/wallet/android/domain/usecases/VerifyDAppUseCase.kt

```diff
@@ -24,14 +25,21 @@ class VerifyDAppUseCase @Inject constructor(
 ) {
 
     suspend operator fun invoke(request: IncomingRequest): Result<Boolean> {
+        val networkId = getProfileUseCase().currentGateway.network.id
+        if (networkId != request.metadata.networkId) {
+            respondToIncomingRequestUseCase.respondWithFailure(
+                request = request,
+                error = DappWalletInteractionErrorType.INVALID_REQUEST
+            )
+            return Result.failure(RadixWalletException.DappRequestException.WrongNetwork(networkId, request.metadata.networkId))
```

Comment:
Can you please add the names of the parameters here? We had a bug in the past were the `networkId` and `request.metadata.networkId` were reversed.